### PR TITLE
New AWS module for ec2 virtual gateway facts

### PR DESCRIPTION
**Pull request by naslanidis**
##### ISSUE TYPE
- New Module Pull Request
##### COMPONENT NAME

ec2_vpc_vgw_facts
##### ANSIBLE VERSION

```
ansible 2.2.0 (devel f39ad1f13a) last updated 2016/05/01 14:01:36 (GMT +1000)
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/usr/share/ansible']
```
##### SUMMARY

This module returns virtual gateway facts for an AWS account.

```
TASK [print output] ************************************************************
ok: [localhost] => {
    "msg": {
        "changed": false, 
        "virtual_gateways": [
            {
                "state": "available", 
                "tags": [
                    {
                        "key": "Name", 
                        "value": "TEST-VGW"
                    }
                ], 
                "type": "ipsec.1", 
                "vpc_attachments": [
                    {
                        "state": "attached", 
                        "vpc_id": "vpc-22a93c74"
                    }
                ], 
                "vpn_gateway_id": "vgw-42e3d64e"
            }
        ]
    }
}
```

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

File: cloud/amazon/ec2_vpc_vgw_facts.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+    type: list
+
+changed:
+    description: True if listing the virtual gateways succeeds
+    type: bool
+    returned: always
+'''
+
+try:
+    import json
```
Comment (wimnat) [inline cloud/amazon/ec2_vpc_vgw_facts.py]: json shouldn't be part of the try

File: cloud/amazon/ec2_vpc_vgw_facts.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    try:
+        region, ec2_url, aws_connect_kwargs = get_aws_connection_info(module, boto3=True)
+        connection = boto3_conn(module, conn_type='client', resource='ec2', region=region, endpoint=ec2_url, **aws_connect_kwargs)
+    except botocore.exceptions.NoCredentialsError, e:
+        module.fail_json(msg="Can't authorize connection - "+str(e))
+
+    # call your function here
+    results = list_virtual_gateways(connection, module)
+    
+    module.exit_json(result=results)
```
Comment (wimnat) [inline cloud/amazon/ec2_vpc_vgw_facts.py:162]: You need to change the boto result to be ansible friendly by changing from CamelCase to snake_case.

Use the helper function 'camel_dict_to_snake_dict'.  See GUIDELINES and ec2_snapshot_facts for example

**Comment by wimnat** [comment]
needs_revision

**Comment by gregdek** [comment]
Thanks @naslanidis for this PR. This PR requires revisions, either because it fails to build or by reviewer request. Please make the suggested revisions. When you are done, please comment with text 'ready_for_review' and we will put this PR back into review.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by naslanidis** [comment]
ready_for_review

**Comment by gregdek** [comment]
Thanks @naslanidis for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

[This message brought to you by your friendly Ansibull-bot.]

**Comment by kalyat** [comment]
We have been using it for months now. shipit

**Comment by ljudzia** [comment]
Been using this it is very useful! shipit

**Pull request closed without merging**